erpnext/stock/doctype/item/item.js

```diff
@@ -121,6 +121,8 @@ frappe.ui.form.on("Item", {
 			frm.set_df_property(fieldname, 'read_only', stock_exists);
 		});
 
+		frm.set_df_property('variants_section', 'hidden', stock_exists);
```

Comment:
this will hide variants section in a variant too if stock exists, and its important for it to be visible there. I think just make the checkbox read only is best

Reply:
ohh okay